documentation/src/test/java/org/hibernate/search/documentation/search/aggregation/AggregationDslIT.java

```diff
@@ -410,6 +410,35 @@ void filter() {
 		} );
 	}
 
+	@Test
+	void withParameters() {
+		withinSearchSession( searchSession -> {
+			// tag::with-parameters[]
+			AggregationKey<Map<Range<Double>, Long>> countsByPriceKey = AggregationKey.of( "countsByPrice" );
+			SearchResult<Book> result = searchSession.search( Book.class )
+					.where( f -> f.matchAll() )
+					.aggregation( countsByPriceKey, f -> f.withParameters( ctx -> ctx.aggregation().range() // <1>
```

Comment:
Not sure exposing the factory here is really necessary... you could just do this:

```suggestion
					.aggregation( countsByPriceKey, f -> f.withParameters( params -> f.range() // <1>
```

And instead of a context, you'd directly pass a `NamedValues` class similar to what's suggested in https://hibernate.atlassian.net/browse/HSEARCH-5075, so that the param retrieval reads `params.get("bound0", Double.class)`.

At least I _think_ this would feel more natural... WDYT?

Reply:
hmm, IDK 😃 🙈, there's nothing stopping the user from doing it right now (meaning ignoring the factory from the context and going with the one from the "outer lambda" 😃) I wasn't sure though if it is ok to do such "nesting" of lambdas. But if this won't be a problem (using `f` from outer lambda), I don't mind making the adjustments 😃 

this got me thinking .... what if we add to `SearchQueryWhereStep` something like:
```java
N where(BiFunction<? super PDF, ? extends NamedValues, ? extends PredicateFinalStep> predicateContributor);
```
and for all other entry points -- projection/sort/aggregation...

then the user could've done something like:
```java
.where( (f, param) -> f.match().field( "title" ).matching( param.get( "myParam" ) ) )
```
the problem is that this kinda clashes with the 
```
N where(BiConsumer<? super PDF, ? super SimpleBooleanPredicateClausesCollector<?>> predicateContributor);
```
(when defining lambdas) 😔

Reply:
> there's nothing stopping the user from doing it right now (meaning ignoring the factory from the context and going with the one from the "outer lambda" 😃)

Sure. I just felt that this would work better with HSEARCH-5075: one would probably rather write `params.get(...)` than `ctx.params().get(...)`.

> I wasn't sure though if it is ok to do such "nesting" of lambdas. But if this won't be a problem (using f from outer lambda), I don't mind making the adjustments 😃

It technically means lambdas would be capturing, but currently any lambda used in the DSL that relies on user input (most likely the vast majority of them) is capturing. So... no big deal, I think.

> this got me thinking .... what if we add to SearchQueryWhereStep something like:

Tempting, but as you just saw it's also not possible ;D Also this basically means the entire predicate tree would be created later, with the downsides we know when it comes to stacktraces.

I think we just can't cover every use case at the root, and because of the kind of clashes you just noticed, we need to be conservative. So I'd wait a bit for the API to mature before thinking about this.

Reply:
```
2024-03-27T16:28:42.1170340Z [ERROR] --[ Constraint Violation ]-----------------------------------------
2024-03-27T16:28:42.1171995Z [ERROR] Constraint: hsearch:PublicTypesMayNotDeclareMethodsWithGetOrSetPrefix
2024-03-27T16:28:42.1173179Z [ERROR] Severity: MAJOR
2024-03-27T16:28:42.1222689Z [ERROR] Number of rows: 1
2024-03-27T16:28:42.1236027Z [ERROR] type=org.hibernate.search.engine.search.common.NamedValues, method=java.util.Optional getOptional(java.lang.String,java.lang.Class)
```
😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 😭 

😆 
so maybe:

```java
public interface NamedValues {
	<T> T get(String name, Class<T> paramType);
	<T> Optional<T> optional(String name, Class<T> paramType);
}
```
m?

Reply:
Rules are made to be broken :)

Try `@SuppressJQAssistant`, and if that doesn't work adjust that specific constraint to take the annotation into account (there are examples in other constraints).